learn-regular-expressions-by-building-a-spam-filter/spamFilter010.js

```diff
@@ -0,0 +1,16 @@
+const isSpam = msg => {
```

Comment:
OK, I'll rework it to take a more gradually incremented approach.

I'm not sure about explaining arrow functions here, though - it doesn't have anything to do with regexes and would probably have already been explained by this point in the curriculum? If it hasn't, we can just use a function declaration - that way we won't end up re-explaining arrow functions in a load of unrelated challenges.

A variable declared with `const` but not assigned throws a SyntaxError. The closest approximation would be `const re = undefined`.

Also, the empty regex is `/(?:)/`, not `//` (the latter is a comment and will throw a SyntaxError if you try to assign it to variable). Perhaps would be better to start with this than `/./`, but then it requires explaining non-capturing groups. Or maybe simply `/a/` or some literal character is the simplest regex conceptually?

Reply:
Very good points here.

About the arrow functions, totally your call! Ideally users will work through the curriculum from beginning to end, but it's possible that they jump around. For cases like that it's good to review, at least briefly. I like the idea of using arrow functions here, and you don't have to go into them in depth. You could show an example and just leave it at that. Using a function declaration is fine, too.

I always forget about `const` throwing a SyntaxError if unassigned. Same with `//`. My mistake! Using `/a/` as a placeholder is a great idea. Mainly we want users to build as much of each project as possible. It's a bit too early in the project to explain non-capturing groups -- which probably aren't required at all -- so going with a single letter is probably best.

Reply:
On second thought, it would be better to start with `const re = /a/;`. It would give you some room to explain how the expression works at this point and would match the first "a" character in a string. Then you could expand on that by introducing the concept of matching whole words like viagra.